tests/model/potential/test_angle.py

```diff
@@ -223,12 +223,22 @@ def test_energy(self):
         u = harmonic_angle.energy(type_=("1"), theta=theta_input)
         self.assertAlmostEqual(u, u_actual)
 
-        # test array r
+        # test invalid theta less than zero
         theta_input = numpy.array([0.0, 0.5, 1.0])
         u_actual = numpy.array([500.0, 125.0, 0.0])
         u = harmonic_angle.energy(type_=("1"), theta=theta_input)
         numpy.testing.assert_allclose(u, u_actual)
 
+        # test invalid theta less than zero
+        theta_input = -1.0
+        with self.assertRaises(ValueError):
+            harmonic_angle.energy(type_=("1"), theta=theta_input)
```

Comment:
FYI: these probably all need a trailing comma
```suggestion
        with self.assertRaises(ValueError):
            harmonic_angle.energy(type_=("1",), theta=-1.0)
```
since otherwise the parentheses are ignored and it's like you've just passed the string. It works here because the string is interpreted as a list of characters, but would be a problem if you had multi-character types.

I would also just put the value of theta in the call, there isn't a reason to separate it out.

Reply:
Putting the trailing comma causes as an error since the key gets set as ("1",). 

>   File "/Users/levipetix/Documents/research/projects/software/relentless/tests/model/potential/test_angle.py", line 234, in test_energy
    harmonic_angle.energy(type_=("1",), theta=-1.0)
  File "/Users/levipetix/Documents/research/projects/software/relentless/src/relentless/model/potential/angle.py", line 117, in energy
    params = self.coeff.evaluate(type_)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/levipetix/Documents/research/projects/software/relentless/src/relentless/model/potential/potential.py", line 73, in evaluate
    if self[key][p] is not None:
       ~~~~^^^^^
  File "/Users/levipetix/Documents/research/projects/software/relentless/src/relentless/model/potential/potential.py", line 96, in __getitem__
    return self._data[key]
           ~~~~~~~~~~^^^^^
  File "/Users/levipetix/Documents/research/projects/software/relentless/src/relentless/collections.py", line 98, in __getitem__
    key = self._check_key(key)
          ^^^^^^^^^^^^^^^^^^^^
  File "/Users/levipetix/Documents/research/projects/software/relentless/src/relentless/collections.py", line 94, in _check_key
    raise KeyError("Key {} is not in dictionary.".format(key))
KeyError: "Key ('1',) is not in dictionary."

Reply:
Oh sorry, you are right: there just shouldn't be any parentheses here!